core/src/mindustry/ClientLauncher.java

```diff
@@ -182,6 +185,14 @@ public void exit(){
 
     @Override
     public void init(){
+        String dataDir = OS.env("MINDUSTRY_DATA_DIR");
+        if(dataDir != null){
+            if(Paths.get(dataDir).isAbsolute()){
+                Core.settings.setDataDirectory(Core.files.absolute(dataDir));
+            }else{
+                throw new IllegalArgumentException("MINDUSTRY_DATA_DIR must be an absolute file path");
```

Comment:
```suggestion
                throw new IllegalArgumentException("MINDUSTRY_DATA_DIR must be an absolute file path.");
```

Reply:
Just in case, this change has been removed because there is no need to enforce paths being absolute, thanks for the contribution!